packages/camera/camera_windows/windows/task_runner_window.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "task_runner_window.h"
+
+#include <algorithm>
+#include <iostream>
+
+namespace camera_windows {
+
+TaskRunnerWindow::TaskRunnerWindow() {
+  WNDCLASS window_class = RegisterWindowClass();
+  window_handle_ =
+      CreateWindowEx(0, window_class.lpszClassName, L"", 0, 0, 0, 0, 0,
+                     HWND_MESSAGE, nullptr, window_class.hInstance, nullptr);
+
+  if (window_handle_) {
+    SetWindowLongPtr(window_handle_, GWLP_USERDATA,
+                     reinterpret_cast<LONG_PTR>(this));
+  } else {
+    auto error = GetLastError();
+    LPWSTR message = nullptr;
+    FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+                       FORMAT_MESSAGE_IGNORE_INSERTS,
+                   NULL, error, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                   reinterpret_cast<LPWSTR>(&message), 0, NULL);
+    OutputDebugString(message);
+    LocalFree(message);
+  }
+}
+
+TaskRunnerWindow::~TaskRunnerWindow() {
+  if (window_handle_) {
+    DestroyWindow(window_handle_);
```

Comment:
Can this use a `unique_ptr` with a custom destructor, instead of managing the reference manually?